Sources/AEPOptimize/Optimize.swift

```diff
@@ -452,9 +471,18 @@ public class Optimize: NSObject, Extension {
             return
         }
 
+        // check if the requested scopes are present in preview cache
+        let previewPropositionDict = previewCachedPropositions.filter { decisionScopes.contains($0.key) }
         let propositionsDict = cachedPropositions.filter { decisionScopes.contains($0.key) }
 
-        let eventData = [OptimizeConstants.EventDataKeys.PROPOSITIONS: propositionsDict].asDictionary()
+        var eventData: [String: Any]?
+        if !previewPropositionDict.isEmpty {
```

Comment:
This looks good! Are we also planning to have a similar check in `processUpdatePropositions` and remove the decisionScope from the edge network request if it exists in `previewCachedPropositions`, to avoid the prod content replacing the preview content?

Reply:
For now, we are not modifying any update workflow and that we can decide after discussion with product. Since we are modifying the get proposition workflow where we check for decisionScope in preview cache, so we are handling the case where even if update proposition call replace the preview content, get will retrieve from preview cache only.